cmd/root.go

```diff
@@ -18,396 +18,436 @@ package cmd
-
-	err := configuration.LoadConfigFromFile(configuration.ConfigLocation, config)
-	if err != nil {
-		LogLady.Info("Unable to load config from file")
-	}
-
-	//err = flag.CommandLine.Parse(args)
-	//if err != nil {
-	//	return config, err
-	//}
-
-	//modfilePath, err := getModfilePath()
-	modfilePath, err := getModfilePathFromCmd(args)
-	if err != nil {
-		return err
-	}
-	if len(modfilePath) > 0 {
-		config.Path = modfilePath
-	} else {
-		config.UseStdIn = true
+func fileExists(filename string) bool {
+	info, err := os.Stat(filename)
+	if os.IsNotExist(err) {
+		return false
 	}
+	return !info.IsDir()
+}
 
-	if outputFormats[outputFormat] != nil {
-		config.Formatter = outputFormats[outputFormat]
-	} else {
+func processConfig() (err error) {
+	switch format := outputFormat; format {
+	case "text":
+		configOssi.Formatter = audit.AuditLogTextFormatter{Quiet: configOssi.Quiet, NoColor: configOssi.NoColor}
+	case "json":
+		configOssi.Formatter = audit.JsonFormatter{}
+	case "json-pretty":
+		configOssi.Formatter = audit.JsonFormatter{PrettyPrint: true}
+	case "csv":
+		configOssi.Formatter = audit.CsvFormatter{Quiet: configOssi.Quiet}
+	default:
 		fmt.Println("!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!")
-		fmt.Println("!!! Output format of", strings.TrimSpace(outputFormat), "is not valid. Defaulting to text output")
+		fmt.Println("!!! Output format of", strings.TrimSpace(format), "is not valid. Defaulting to text output")
 		fmt.Println("!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!")
-		config.Formatter = outputFormats["text"]
+		configOssi.Formatter = audit.AuditLogTextFormatter{Quiet: configOssi.Quiet, NoColor: configOssi.NoColor}
 	}
 
-	err = configuration.GetCVEExcludesFromFile(config, excludeVulnerabilityFilePath)
-	if err != nil {
-		return err
+	switch configOssi.LogLevel {
+	case 1:
+		logLady.Level = logrus.InfoLevel
+	case 2:
+		logLady.Level = logrus.DebugLevel
+	case 3:
+		logLady.Level = logrus.TraceLevel
 	}
 
-	return nil
-}
+	ossIndex := ossiCreator.create()
+
+	if configOssi.CleanCache {
+		logLady.Info("Attempting to clean cache")
+		if err = ossIndex.NoCacheNoProblems(); err != nil {
+			logLady.WithField("error", err).Error("Error cleaning cache")
+			fmt.Printf("ERROR: cleaning cache: %v\n", err)
+			return
+		}
+		logLady.Info("Cache cleaned")
+		return
+	}
 
-func getModfilePathFromCmd(args []string) (modfilepath string, err error) {
-	if len(args) > 0 {
-		if len(args) != 1 {
-			return modfilepath, fmt.Errorf("wrong number of modfile paths: %s", args)
+	printHeader(!configOssi.Quiet && reflect.TypeOf(configOssi.Formatter).String() == "audit.AuditLogTextFormatter")
+
+	// todo: should errors from this call be ignored
+	_ = getCVEExcludesFromFile(excludeVulnerabilityFilePath)
```

Comment:
My vote on it is yeah it probably should error out If it bombs people should know about it since it could cause some churn trying to figure out why excludes didn't work and we havent even logged anything out.